otel/otel.go

```diff
@@ -130,25 +132,34 @@ func NewOpenTelemetry(ctx context.Context, config *OTelConfig, logger *zap.Logge
 	var shutdownFuncs []func(context.Context) error
 	resource := otelInst.createResource(ctx)
 
-	// Initialize TracerProvider
-	otelInst.TracerProvider, err = otelInst.InitTracerProvider(ctx, resource)
-	if err != nil {
-		logger.Error("error while initializing the tracer provider", zap.Error(err))
-		return nil, nil, err
+	if config.TraceEnabled {
+		// Initialize TracerProvider
+		otelInst.TracerProvider, err = otelInst.InitTracerProvider(ctx, resource)
+		if err != nil {
+			logger.Error("error while initializing the tracer provider", zap.Error(err))
+			return nil, nil, err
+		}
+		otel.SetTracerProvider(otelInst.TracerProvider)
+		otelInst.Tracer = otelInst.TracerProvider.Tracer(config.ServiceName)
```

Comment:
line 143 and 146 is essentially same. We can rewrite this as 
if (config.TraceEnabled) {
  // Code to initialize and set otel.SetTraceProvider
}
// common code to initialize otelInst.Tracer

Reply:
sure, done